### PR TITLE
Add Array.mapi_inplace, Float.Array.map_inplace, Float.Array.mapi_inplace

### PR DESCRIPTION
**Pull request by redianthus**
Hi,

As suggested in #11836 by @alainfrisch, I also added `Array.mapi_inplace`.

### CODE REVIEW

**Comment by nojb** [comment]
One suggestion that I forgot to make in #11836: I think it would be a good idea to add `map_inplace` and `mapi_inplace` to `Float.Array` as well (as I see it, the signatures of `Array` and `Float.Array` should be kept in lockstep as much as possible).

File: stdlib/array.mli [hunk 1]
```diff
@@ -170,6 +170,11 @@ val mapi : (int -> 'a -> 'b) -> 'a array -> 'b array
    function is applied to the index of the element as first argument,
    and the element itself as second argument. *)
 
+val mapi_inplace : (int -> 'a -> 'a) -> 'a array -> unit
+(** Same as {!map_inplace}, but the function is applied to the index of the
+    element as first argument, and the element itserf as second argument.
```
Comment (nojb) [inline stdlib/array.mli]:
```suggestion
    element as first argument, and the element itself as second argument.
```

File: stdlib/arrayLabels.mli [hunk 2]
```diff
@@ -170,6 +170,11 @@ val mapi : f:(int -> 'a -> 'b) -> 'a array -> 'b array
    function is applied to the index of the element as first argument,
    and the element itself as second argument. *)
 
+val mapi_inplace : f:(int -> 'a -> 'a) -> 'a array -> unit
+(** Same as {!map_inplace}, but the function is applied to the index of the
+    element as first argument, and the element itserf as second argument.
```
Comment (nojb) [inline stdlib/arrayLabels.mli]:
```suggestion
    element as first argument, and the element itself as second argument.
```

File: stdlib/array.mli (hunk 1 above)
Comment (redianthus) [reply stdlib/array.mli]: Fixed.

File: stdlib/arrayLabels.mli (hunk 2 above)
Comment (redianthus) [reply stdlib/arrayLabels.mli]: Fixed.

**Review by nojb — APPROVED** [review]
LGTM. As this PR touches the standard library, a second approval is needed.

**Comment by nojb** [comment]
@gasche: as you reviewed #11836, do you mind giving this one a look? Thanks!

File: Changes [hunk 3]
```diff
@@ -72,6 +72,9 @@ Working version
 
 ### Standard library:
 
+- #11837: Add `Array.mapi_inplace`.
+  (Léo Andrès, review by ?)
```
Comment (gasche) [inline Changes]: You could fold this entry in the one below (#..., #...: Array.map_inplace, Array.mapi_inplace), and add Nicolás as a reviewer there.

**Review by gasche — APPROVED** [review]
This looks reasonable to me, approved.

File: Changes (hunk 3 above)
Comment (nojb) [reply Changes]: Good idea, and also mention `Float.Array.map_inplace`, `Float.Array.mapi_inplace`.

**Comment by nojb** [comment]
@zapashcanon: the PR needs to be rebased on the latest `trunk` following #11858.

**Comment by redianthus** [comment]
@nojb yes, I was rebasing when you replied. It should be good now.

**Comment by nojb** [comment]
> @nojb yes, I was rebasing when you replied. It should be good now.

Thanks! This can be merged if the CI passes.

**Pull request merged**